File: lab06/matrix.cpp
#include <iostream>
#include <string>
#include <math.h>

using namespace std;

int main()
{
        cout << "Welcome to Matrix Multiplication!" << endl;

        int matSize;
	int sum = 0;

        cout << "Enter size of Matrix:" << endl;
        cin >> matSize;

        int A[matSize][matSize], B[matSize][matSize], C[matSize][matSize];

        for(int n=0; n < 2; n++){
                for(int i=0; i < matSize; i++){
                        for(int j=0; j < matSize; j++){
                                if(n == 0){
                                        cout << "Enter value for position A[" << i << "][" << j << "]" << endl;
                                        cin >> A[i][j];
                                } else {
                                        cout << "Enter value for position B[" << i << "][" << j << "]" << endl;
                                        cin >> B[i][j];
                                }
                        }
                }
        }

	for(int i=0; i < matSize; i++){
		for(int j=0; j < matSize; j++){
			C[i][j] = 0;
			for(int n = 0; n < matSize; n++){
				C[i][j] += A[i][n] * B[n][j];
			}
		}
	}
	
	for(int i=0; i < matSize; i++){
        	for(int j=0; j < matSize; j++){
                	cout << "C[" << i << "][" << j << "] = " << C[i][j] << endl;
		}
	}

	return 0;
}

File: lab07/matrix.py
#!/bin/python3

print("Welcome to Matrix Multiplication!")

matSize = int(input("Enter size of Matrix: "))

A = [[0 for col in range(matSize)] for row in range(matSize)]
B = [[0 for col in range(matSize)] for row in range(matSize)]
C = [[0 for col in range(matSize)] for row in range(matSize)]

for n in range(0,2):
    for i in range(0,matSize):
        for j in range(0,matSize):
            if n == 0:
                aVal = input("Enter the value for position A[%d][%d]:\n" % (i,j))
                A[i][j] = int(aVal)
            else:
                bVal = input("Enter the value for position B[%d][%d]:\n" % (i,j))
                B[i][j] = int(bVal)

for i in range(0,matSize):
    for j in range(0,matSize):
        C[i][j] = 0
        for n in range(0,matSize):
            C[i][j] += A[i][n] * B[n][j]

for i in range(0,matSize):
    for j in range(0,matSize):
        print("C[%d][%d] = %d" % (i,j,C[i][j]))
